src/main.rs

```diff
@@ -41,18 +42,23 @@ fn main() {
 }
 
 /// Try to read `shell.nix` from the current working dir.
-fn get_shell_nix() -> Result<NixFile, ExitError> {
+fn get_shell_nix(shellfile: PathBuf) -> Result<NixFile, ExitError> {
     let current_dir_msg = || match env::current_dir() {
         Err(_) => String::from(""),
         Ok(pb) => format!(" ({})", pb.display()),
     };
     // use shell.nix from cwd
-    Ok(NixFile::from(locate_file::in_cwd("shell.nix").map_err(
+    let shellfile_str = match shellfile.to_str() {
+        Some(i) => i,
+        _ => "",
+    };
+    Ok(NixFile::from(locate_file::in_cwd(shellfile_str).map_err(
```

Comment:
`shellfile.to_str()` might return an error if the given string is not valid UTF-8. We should change `locate_file::in_cwd()` to work on `PathBuf`s or `OSString` instead (so we don’t have to decode to UTF-8 unnecessarily).
If you don’t know how to do that, I can do it afterwards if you want.

Reply:
Should be done now. 
But we still need to use shellfile.to_str() to show the failing path (in case of error)

Reply:
You can use https://doc.rust-lang.org/std/path/struct.Path.html#method.display, which is defined for `Path` and thus also for `PathBuf` (`Path` is the borrowed version of `PathBuf`).

Reply:
Learnt something, thank you. Should be fixed now.